strapi/strapi
# Add mysql2 as a db config client option

### What does it do?

Adds `mysql2` as a valid client project in database configurations

My first attempt was [a different approach that maintains client='mysql' in database configuration, but attempts to rewrite it only for knex](https://github.com/strapi/strapi/pull/16065) and tries to choose that client based on package availability, whereas this PR avoids the complication and instead allows 'mysql2' as a valid client option in the database configuration. It also leaves the implementation of sqlite packages untouched to avoid unintentionally breaking anything.

I prefer this approach over the original PR because it's more flexible for users to be able to choose their own knex client, more clear what is happening (less magic behind the scenes), and also more "future proof" in case new packages could be added in the future.

### Why is it needed?

To allow mysql2 as the knex driver, which adds support for Mysql 8 authentication (among other improvements)

### How to test it?

- Create a Strapi project
- yarn add mysql2
- in database configuration, set client to 'mysql2' to begin using the mysql2 package
- should now work with mysql 8

### TODO

- [x] ~~investigate if packages/plugins/i18n/server/migrations/field/migrate-for-bookshelf.js is receiving the actual client name ('mysql2' would cause the bookshelf migration to fail) or if it is receiving the dialect name ('mysql') which would be fine.~~ this code is from v3 and will be unaffected

### Related issue(s)/PR(s)

[A different approach that maintains client='mysql' but attempts to rewrite it only for knex](https://github.com/strapi/strapi/pull/16065)

Fixes https://github.com/strapi/strapi/issues/13774

Initial PR fix from community user: https://github.com/strapi/strapi/pull/13775
(Missing update to tests, but credit to @benderillo for getting the ball rolling)

[Documentation PR](https://github.com/strapi/documentation/pull/1592) -- was written for the original PR, if this one is accepted instead I will update the docs to reflect it
